File: src/rs/p1.rs
mod operations {

    /* Sums all numbers in given range */
    pub fn sumall(of: u32) -> u32 {
        if of > 0 {
            of + sumall(of - 1)
        } else {
            0
        }
    }


    /* Counts factorial of given number */
    pub fn factorial(of: u32) -> u32 {
        if of > 1 {
            of * factorial(of - 1)
        } else {
            1
        }
    }


    /* sum list of values v1 */
    pub fn sum_list1(vector: &Vec<usize>) -> usize {
        vector
            .iter()
            .fold(0, |value, accum| value + *accum)
    }


    /* sum list of values v2 */
    pub fn sum_list2(vector: &Vec<usize>) -> usize {
        vector
            .iter()
            .sum()
    }

}


fn main() {
    use operations::*;

    // let of = 0;
    let of = 10;

    /* 1st call */
    println!("Rust: {}! = {}, sumall = {}",
             of, factorial(of), sumall(of));
    /* Nth call */
    println!("Rust: 0! = {}, sumall = {}",
             factorial(0), sumall(0));

    let a_list = vec!(1,2,3,4,5,6,7,8,9,10);
    println!("a_list1: {:?}", &a_list);
    println!("Sum_list1: {}, Sum_list2: {}",
             sum_list1(&a_list), sum_list2(&a_list));
}

File: src/exs/p1.exs
defmodule Operations do


  # Sums all numbers in given range
  def sumall(of) when is_number(of) do
    if of > 0 do
      of + sumall(of - 1)
    else
      0
    end
  end


  # Counts factorial of given number
  def factorial(of) when is_number(of) do
    cond do
      of > 1 ->
        of * factorial(of - 1)
      of <= 1 ->
        1
    end
  end


end


defmodule Main do


  def main do
    import Operations

    # of = 9
    of = 10
    sum_of = sumall of
    factorial_of = factorial of

    # 1st call:
    IO.puts "Elixir: #{ of }! = #{ factorial_of }, sumall = #{ sum_of }"

    # Nth call:
    IO.puts "Elixir: 0! = #{ factorial(0) }, sumall = #{ sumall(0) }"
  end


end


Main.main()
